src/utils/files.ts

```diff
@@ -33,6 +41,147 @@ export const exists = async (filePath: string): Promise<boolean> =>
+ * to be issues with using Archiver.js with Electron/Webpack,
+ * so that's why we have to do it in a somewhat inefficient way.
+ *
+ * Related issues:
+ *  * https://github.com/archiverjs/node-archiver/issues/349
+ *  * https://github.com/archiverjs/node-archiver/issues/403
+ *  * https://github.com/archiverjs/node-archiver/issues/174
+ *
+ * @param archive ZIP archive to add the file to
+ * @param filePath file to add, absolute path
+ * @param nameInArchive name of file in archive
+ */
+const addFileToZip = async (
+  archive: archiver.Archiver,
+  filePath: string,
+  nameInArchive: string,
+) => {
+  return archive.append(await fsPromises.readFile(filePath), { name: nameInArchive });
+};
+
+// Generate a sequence of all regular files inside the given directory
+async function* getFiles(dir: string): AsyncGenerator<string> {
+  const dirents = await fsPromises.readdir(dir, { withFileTypes: true });
+  for (const dirent of dirents) {
+    const res = path.resolve(dir, dirent.name);
+    if (dirent.isDirectory()) {
+      yield* getFiles(res);
+    } else if (dirent.isFile()) {
+      yield res;
+    }
+  }
+}
+
+/**
+ * Add the given path to the archive. If it's a file we add it directly, it it is a directory
+ * we recurse over all the files within that directory
+ *
+ * @param archive ZIP archive to add the file to
+ * @param filePath file to add, absolute path
+ */
+const addFileOrDirectoryToZip = async (archive: archiver.Archiver, filePath: string) => {
+  const pathExists = await exists(filePath);
+  if (!pathExists) {
+    throw Error(`cannot zip nonexistant path: ${filePath}`);
+  }
+
+  const isDir = await fsPromises.lstat(filePath).then(res => res.isDirectory());
+  if (isDir) {
+    log.info('Adding directory to zip file:', filePath);
+    for await (const file of getFiles(filePath)) {
+      // a typical file might look like this:
+      //   /home/user/.polar/networks/1/volumes/bitcoind/backend1/regtest/mempool.dat
+      // after applying this transformation, we end up with:
+      //   volumes/bitcoind/backend1/regtest/mempool.dat
+      const nameInArchive = path.join(
+        path.basename(filePath),
+        file.slice(filePath.length),
+      );
+      await addFileToZip(archive, file, nameInArchive);
+    }
+  } else {
+    return addFileToZip(archive, filePath, path.basename(filePath));
+  }
+};
+
+/**
+ * Archive the given network into a folder with the following content:
+ *
+ * ```
+ * docker-compose.yml // compose file for network
+ * volumes            // directory with all data files needed by nodes
+ * network.json       // serialized network object
+ * chart.json         // serialized chart object
+ * ```
+ *
+ * @return Path of created `.zip` file
+ */
+export const zipNetwork = async (network: Network, chart: IChart): Promise<string> => {
+  return new Promise(async (resolve, reject) => {
+    const zipPath = join(os.tmpdir(), `polar-${network.name}.zip`);
```

Comment:
Should sanitize the network name since it could contain special characters that aren't allowed in file names (ex: /, ?, <, >, \, :, *, etc.)